src/app.js

```diff
@@ -1,7 +1,20 @@
 /* @flow */
 'use strict'
 
+// SIGINT and uncaughtException do not run cleanup
+// so we have an exitHandler to clean up the database connection
+function exitHandler (error) {
+  if (error) {
+    console.error('Exiting: ', error)
+    process.exit(1)
+  }
+  process.exit(0)
+}
+
 module.exports = require('./lib/app')
 if (!module.parent) {
+  process.on('SIGINT', exitHandler)
+  process.on('uncaughtException', exitHandler)
```

Comment:
>  I'm trying to understand more about the problem that you're trying to solve, maybe there is a less hacky approach.

This is the problem we're solving. SIGINT is sent by process manager to the node process when it wants to shut it down. Knex maintains a connection pool, which does not close automatically on a SIGINT. So we have to trap that signal, and exit the process, which forces the connection to close.
See:
https://github.com/tgriesser/bookshelf/issues/405

`throw null` is a bug in javascript it seems. Not sure how we'd get around that.